### PR TITLE
initial IANA media type registration

### PR DESCRIPTION
**Pull request by Sakurann**
if merged resolves Issue #74.

### CODE REVIEW

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 1]
```diff
@@ -880,7 +880,60 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
 
 # IANA Considerations {#iana_considerations}
```
Comment (bc-pi) [inline draft-ietf-oauth-selective-disclosure-jwt.md:881]: We probably should keep some kind of TBD in here as a reminder for other things - like JWT claims - that'll need registration.

Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:881]:
```suggestion
# IANA Considerations {#iana_considerations}

TBD
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 2]
```diff
@@ -880,7 +880,60 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
...
 
-TBD
+## Media Type Registration
+
+This section registers the "application/sdjwt-issuance" media type [@RFC2046] in
```
Comment (bc-pi) [inline draft-ietf-oauth-selective-disclosure-jwt.md:885]: I hate to bikeshed but SD-JWT is hyphenated everywhere else in this draft so `sdjwt-issuance` and `sdjwt-presentation` look a bit off.  Is there a reason not to go w/ `sd-jwt-issuance` and `sd-jwt-presentation`?

Comment (bc-pi) [reply draft-ietf-oauth-selective-disclosure-jwt.md:885]: or if length is of concern, something like `sd-jwt-i` and `sd-jwt-p`

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 3]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
...
+in the "Media Types" registry, which can be used to indicate that
+the content is a Combined Format for Presentation.
+
+### application/sdjwt-issuance
+
+*  Type name: application
+*  Subtype name: sdjwt-issuance
+*  Required parameters: n/a
+*  Optional parameters: n/a
+*  Encoding considerations: 8bit; application/sdjwt-issuance values are encoded as a series of base64url-encoded values, each separated from the next by a single tilde ('~') character.
```
Comment (bc-pi) [inline draft-ietf-oauth-selective-disclosure-jwt.md:900]: I don't know how specific the  Encoding considerations needs to be but the dot character also is a separator in the JWT part.

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 4]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
...
+*  Change Controller: IESG
+*  Provisional registration?  No
+
+### application/sdjwt-presentation
+
+*  Type name: application
+*  Subtype name: sdjwt-presentation
+*  Required parameters: n/a
+*  Optional parameters: n/a
+*  Encoding considerations: 8bit; application/sdjwt-presentation values are encoded as a series of base64url-encoded values (the last of which may be an empty string), each separated from the next by a single tilde ('~') character.
```
Comment (bc-pi) [inline draft-ietf-oauth-selective-disclosure-jwt.md:923]: dot character will be a separator too ...

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 2 above)
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:887]: Please bikeshed :)

```suggestion
This section registers the "application/sd-jwt-issuance" media type [@RFC2046] in
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 5]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
 
 TBD
 
+## Media Type Registration
+
+This section registers the "application/sdjwt-issuance" media type [@RFC2046] in
+the "Media Types" registry [@IANA.MediaTypes] in the manner described
+in [@RFC6838], which can be used to indicate that the content is a Combined Format for Issuance.
+This section also registers the "application/sdjwt-presentation" media type
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:890]:
```suggestion
This section also registers the "application/sd-jwt-presentation" media type
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 6]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
...
+in the "Media Types" registry, which can be used to indicate that
+the content is a Combined Format for Presentation.
+
+### application/sdjwt-issuance
+
+*  Type name: application
+*  Subtype name: sdjwt-issuance
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:897]:
```suggestion
*  Subtype name: sd-jwt-issuance
```

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 3 above)
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:900]:
```suggestion
*  Encoding considerations: 8bit; application/sd-jwt-issuance values are encoded as a series of base64url-encoded values, each separated from the next by a single tilde ('~') character.
```

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 4 above)
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:923]:
```suggestion
*  Encoding considerations: 8bit; application/sd-jwt-presentation values are encoded as a series of base64url-encoded values (the last of which may be an empty string), each separated from the next by a single tilde ('~') character.
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 7]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
...
+## Media Type Registration
+
+This section registers the "application/sdjwt-issuance" media type [@RFC2046] in
+the "Media Types" registry [@IANA.MediaTypes] in the manner described
+in [@RFC6838], which can be used to indicate that the content is a Combined Format for Issuance.
+This section also registers the "application/sdjwt-presentation" media type
+in the "Media Types" registry, which can be used to indicate that
+the content is a Combined Format for Presentation.
+
+### application/sdjwt-issuance
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:894]:
```suggestion
### application/sd-jwt-issuance
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 8]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
...
+      File extension(s): n/a
+      Macintosh file type code(s): n/a
+*  Person & email address to contact for further information: Daniel Fett, mail@danielfett.de
+*  Intended usage: COMMON
+*  Restrictions on usage: none
+*  Author: Daniel Fett, mail@danielfett.de
+*  Change Controller: IESG
+*  Provisional registration?  No
+
+### application/sdjwt-presentation
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:917]:
```suggestion
### application/sd-jwt-presentation
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 9]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
...
+
+*  Type name: application
+*  Subtype name: sdjwt-presentation
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:920]:
```suggestion
*  Subtype name: sd-jwt-presentation
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 10]
```diff
@@ -882,6 +882,61 @@ The work on this draft was started at OAuth Security Workshop 2022 in Trondheim,
 
 TBD
 
+## Media Type Registration
+
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:886]:
```suggestion

### Registry Contents

```

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 3 above)
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:900]:
```suggestion
*  Encoding considerations: 8bit; application/sd-jwt-issuance values are encoded such that the first three base64url-encoded values are separated from the previous one by a single period ('.') character, and the following series of base64url-encoded values are each separated from the next by a single tilde ('~') character.
```

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 4 above)
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:925]:
```suggestion
*  Encoding considerations: 8bit; application/sdjwt-presentation values are encoded as a series of base64url-encoded values (the last of which may be an empty string), each separated from the next by a single tilde ('~') character.
*  Encoding considerations: 8bit; application/sd-jwt-presentation values are encoded such that the first three base64url-encoded values are separated from the previous one by a single period ('.') character, and the following series of base64url-encoded values (the last of which may be an empty string or three base64url-encoded values separated from the next one by a single period ('.') character) are each separated from the previous one by a single tilde ('~') character.
```

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 3 above)
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:902]: I tried.

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 4 above)
Comment (Sakurann) [reply draft-ietf-oauth-selective-disclosure-jwt.md:925]: my best attempt to put it in one sentence

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 3 above)
Comment (danielfett) [reply draft-ietf-oauth-selective-disclosure-jwt.md:902]:
```suggestion
*  Encoding considerations: 8bit; application/sdjwt-issuance values are encoded as a series of strings separated from the next by a single tilde ('~') character; the first string consists of dot-separated base64url-encoded values, all further strings are base64url-encoded values.
```

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 4 above)
Comment (danielfett) [reply draft-ietf-oauth-selective-disclosure-jwt.md:925]:
```suggestion
*  Encoding considerations: 8bit; application/sdjwt-presentation values are encoded as a series of strings, each separated from the next by a single tilde ('~') character; the first and last strings consist of dot-separated base64url-encoded values, all other strings are base64url-encoded values; the last string may be empty.
```

**Review by danielfett — CHANGES_REQUESTED** [review]

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 3 above)
Comment (bc-pi) [reply draft-ietf-oauth-selective-disclosure-jwt.md:902]:
```suggestion
*  Encoding considerations: 8bit; application/sdj-wt-issuance values are a series of base64url-encoded values (some of which may be the empty string) separated by period ('.') or tilde ('~') characters.
```

This tries to simplify it and make it less specific.

**Comment by Sakurann** [comment]
the alternative PR merged

**Pull request closed without merging**